### DISCUSSION TITLE
It would be nice to adapt the code to work with other Languages.

### DISCUSSION THREAD

**Discussion by LucianoSP** [daveshap/Medical_Intake#2] [Ideas]
This is specially useful for developing countries that don´t speak English. The conversation with the API already works in other languages but the reports don´t.

**Comment by daveshap** [comment]
Yeah someone recommended using a global setting. I think this would be the best first issue.

**Comment by solyarisoftware** [comment]
Maybe the language localization is not a big issue, at first glance/approximation.

Assuming that we do have a powerful LLM (e.g. GPT3.5 and successive) capable of speaking target language X (this is a prerequisite), at first approximation you can continue to use English as the reference language. 

> By the way in case the language model wouldn't support language X. In this case you need a I/O language translator as wrapper to the entire application, but honestly recent LLMs support an vast spectrum of human languages. 

The only required modification in prepared prompts is to insert a one-line rule like

```
LANGUAGE: Conduct the conversation in fluent {LANGUAGE}.
```
So you can foresee a global setting variable LANGUAGE set to the target localization language (e.g. equal to `Italian`). 
I deepened this a bit in my article: https://convcomp.it/non-english-languages-prompt-engineering-trade-offs-7e529866faba 

To be picky, the above solution has pros and cons: 
- pro: 
   - long story short: using English minimize tokens (and so costs and latency!) 
   - all "code base" (prompts) remain unique using English
   - Efficiency in ‘Procedural Understanding’ (see my article)
- con: 
  - The English language 'guides' the LLM to adhere to certain English language 'pragmatics' and specific English language and social conventions. Hence, it may be beneficial, from a sociolinguistic perspective, to explore the use of a different native language, although this would require further investigation and validation. In my limited experience with LLMs, this aspect remains somewhat unclear and would need to be scientifically examined through linguistic studies. However, when it comes to the specific application in the healthcare and medical domains, I believe that using English is more than sufficient. The situation might become 'more nuanced' in the case of specialized medical fields such as psychology and psychotherapy, where cultural localization could indeed be relevant
